src/Document.php

```diff
@@ -71,57 +71,73 @@ public function toDomDocument()
     }
     
     /**
-     * Create "Ad" section on "VAST" node
+     * Create inline Ad section
      *
-     * @param string $type
+     * @return \Sokil\Vast\Ad\InLine
+     */
+    public function createInLineAdSection(\DOMElement $domEl = null)
+    {
+        return $this->createAdSection($this->createAdSectionDomEl('InLine'));
+    }
+    
+    /**
+     * Create Wrapper Ad section
      *
-     * @throws \InvalidArgumentException
+     * @return \Sokil\Vast\Ad\Wrapper
+     */
+    public function createWrapperAdSection()
+    {
+        return $this->createAdSection(
+            $this->createAdSectionDomEl('Wrapper')
+        );
+    }
+
+    /**
+     * Create ad node and ad section node of given type
+     *
+     * @param string $type inline|wrapper
      *
-     * @return AbstractAdNode
+     * @return \DOMElement
      */
-    private function createAdSection($type)
+    private function createAdSectionDomEl($type)
     {
-        // Check Ad type
-        $adTypeClassName = '\\Sokil\\Vast\\Ad\\' . $type;
-        if (!class_exists($adTypeClassName)) {
-            throw new \InvalidArgumentException(sprintf('Ad type %s not supported', $type));
-        }
-        
         // create dom node
         $adDomElement = $this->domDocument->createElement('Ad');
         $this->domDocument->documentElement->appendChild($adDomElement);
 
         // create type element
         $adTypeDomElement = $this->domDocument->createElement($type);
         $adDomElement->appendChild($adTypeDomElement);
-        
-        // create ad section
-        $adSection = new $adTypeClassName($adDomElement);
-        
-        // cache
-        $this->vastAdNodeList[] = $adSection;
-        
-        return $adSection;
+
+        return $adTypeDomElement;
     }
-    
+
     /**
-     * Create inline Ad section
+     * Create ad section object from given DOM node
      *
-     * @return \Sokil\Vast\Ad\InLine
-     */
-    public function createInLineAdSection()
-    {
-        return $this->createAdSection('InLine');
-    }
-    
-    /**
-     * Create Wrapper Ad section
+     * @param \DOMElement $domEl
      *
-     * @return \Sokil\Vast\Ad\Wrapper
+     * @return \Sokil\Vast\Ad\InLine|\Sokil\Vast\Ad\Wrapper
+     * @throws \Exception
      */
-    public function createWrapperAdSection()
+    protected function createAdSection(\DOMElement $domEl)
     {
-        return $this->createAdSection('Wrapper');
+        // create ad section
+        switch (\strtolower($domEl->tagName)) {
+            case 'wrapper':
+                $adSection = new \Sokil\Vast\Ad\Wrapper($domEl);
+                break;
+            case 'inline':
+                $adSection = new \Sokil\Vast\Ad\InLine($domEl);
+                break;
+            default:
+                throw new \Exception('Ad type ' . $domEl->tagName . ' not supported');
```

Comment:
library must throw own exception. Maybe better to implement `\Sokil\Vast\Exception` and throw it? And i think we need to move in v.2.0 to some other vendor name ))))